packages/expo-network/src/Network.ts

```diff
@@ -1,9 +1,13 @@
-import { UnavailabilityError } from 'expo-modules-core';
+import { EventEmitter, type Subscription, UnavailabilityError } from 'expo-modules-core';
+import { useEffect, useState } from 'react';
 
 import ExpoNetwork from './ExpoNetwork';
-import { NetworkState, NetworkStateType } from './Network.types';
+import { NetworkState, NetworkStateEvent, NetworkStateType } from './Network.types';
 
-export { NetworkState, NetworkStateType };
+export { NetworkState, NetworkStateEvent, NetworkStateType };
+
+const emitter = new EventEmitter(ExpoNetwork);
```

Comment:
As of SDK 51 we no longer need to use additional `EventEmitter` to send events. The native module object (`ExpoNetwork` here) already inherits from a new emitter class that is written in C++, see:
- type declaration: https://github.com/expo/expo/blob/main/packages/expo-modules-core/src/ts-declarations/EventEmitter.ts
- new implementation: https://github.com/expo/expo/blob/main/packages/expo-modules-core/common/cpp/EventEmitter.cpp

This also solves a long-standing problem with RN's event emitters which are actually global events – event names from different modules can conflict with each other, thus we used to prefix them with something expo-specific.

Reply:
Thanks for the hint, I have adjusted it. However, it would be good if this change was also reflected in the [documentation](https://docs.expo.dev/modules/native-module-tutorial/#typescript-module-1) and in the template (`npx create-expo-module my-module`).

Reply:
Yeah, we're planning to do this, however:
- We want 'create-expo-module` to be compatible with at least one SDK version older than the latest (actually we're going to version them but that's another topic).
- Our docs generation is not yet adjusted to present class inheritance and supported events. We'll also need a new page in the docs that includes all the public APIs exported by the `expo` package (public and stable APIs from `expo-modules-core` should be re-exported by `expo`).

All of those should be solved before SDK 52 though.